src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/DownloadResourceProgressTests.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests.TestAppScenarios;
+
+public class DownloadResourceProgressTests : AppTestBase
+{
+    public DownloadResourceProgressTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task DownloadProgressFinishes(bool fetchFailure)
```

Comment:
Maybe .. `fetchFailure` -> `checkFailingFetches`, or `checkFailedFetches`?

Reply:
I picked `failFirstAssemblyDownload`

Reply:
Failing only the *first* assembly is very specific, and special. Instead, why not fail a fixed number (say 5), but fail 5 random ones on every run?

Reply:
Note we have similar re-try test in the advanced sample

Reply:
IIUC, that's a sample which needs to be run manually to see the results of retry. Maybe then add a wrapper around that sample that can poke at the various points via playwright to make it a test.

Reply:
In general, I would like to have something between samples and tests, something that serves both purposes. Blazor has lot of tests based on BasicTestApp and others, which has a basic UI, where you can poke to things, but it is also the app for tests. I started with WasmBasicTestApp, it can't serve the sample part yet, because we don't have Wasm SDK in tree testing setup yet, but I would like to convert there.

Reply:
I have added more failures